dss/events/handlers/index.py

```diff
@@ -1,36 +1,48 @@
+
+"""Lambda function for DSS indexing"""
+
 import json
 import os
 import re
 from urllib.parse import unquote
 
 import boto3
+import botocore
 
 from ... import DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE
-from ...util import connect_elasticsearch
+from ...hcablobstore import BundleMetadata, BundleFileMetadata
+from ...util import connect_elasticsearch, create_blob_key
 
 DSS_BUNDLE_KEY_REGEX = r"^bundles/[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}\..+$"
 
-#
-# Lambda function for DSS indexing
-#
+
+class ElasticsearchClient:
+    _es_client = None
+
+    @staticmethod
+    def get(logger):
+        if ElasticsearchClient._es_client is None:
+            ElasticsearchClient._es_client = connect_elasticsearch(os.getenv("DSS_ES_ENDPOINT"), logger)
+        return ElasticsearchClient._es_client
 
 
 def process_new_indexable_object(event, logger) -> None:
     try:
         # This function is only called for S3 creation events
         key = unquote(event['Records'][0]["s3"]["object"]["key"])
         if is_bundle_to_index(key):
-            logger.info("Received S3 creation event for bundle which will be indexed: %s", key)
+            logger.info(f"Received S3 creation event for bundle which will be indexed: {key}")
             s3 = boto3.resource('s3')
             bucket_name = event['Records'][0]["s3"]["bucket"]["name"]
             manifest = read_bundle_manifest(s3, bucket_name, key, logger)
-            index_data = create_index_data(s3, bucket_name, manifest, logger)
-            add_index_data_to_elasticsearch(os.getenv("DSS_ES_ENDPOINT"), key, index_data, logger)
-            logger.debug("Finished index processing of S3 creation event for bundle: %s", key)
+            bundle_id = get_bundle_id_from_key(key)
+            index_data = create_index_data(s3, bucket_name, bundle_id, manifest, logger)
+            add_index_data_to_elasticsearch(bundle_id, index_data, logger)
+            logger.debug(f"Finished index processing of S3 creation event for bundle: {key}")
         else:
-            logger.debug("Not indexing S3 creation event for key: %s", key)
-    except Exception as e:
-        logger.error("Exception occurred while processing S3 event: %s Event: %s", e, json.dumps(event, indent=4))
+            logger.debug(f"Not indexing S3 creation event for key: {key}")
+    except Exception as ex:
+        logger.error(f"Exception occurred while processing S3 event: {ex} Event: {json.dumps(event, indent=4)}")
```

Comment:
I think as a project policy we should avoid putting expressions in f-strings. Things can quickly get out of hand otherwise, at minimum with linting and when user-controlled data is concerned.